pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,278 @@
+
+			codec := tester.GetCodec(t)
+			actualEncoding, err := codec.Encode(ctx, items, TestItemArray2Type)
+
+			require.NoError(t, err)
+			assert.Equal(t, resp, actualEncoding)
+
+			var into [2]TestStruct
+			require.NoError(t, codec.Decode(ctx, actualEncoding, &into, TestItemArray2Type))
+			assert.Equal(t, items, into)
+		},
+		"Encodes and decodes an arrays with one element": func(t *testing.T) {
+			item1 := CreateTestStruct(0, tester)
+			items := [1]TestStruct{item1}
+			req := &EncodeRequest{TestStructs: items[:], TestOn: TestItemArray1Type}
+			resp := tester.EncodeFields(t, req)
+
+			codec := tester.GetCodec(t)
+			actualEncoding, err := codec.Encode(ctx, items, TestItemArray1Type)
+
+			require.NoError(t, err)
+			assert.Equal(t, resp, actualEncoding)
+
+			var into [1]TestStruct
+			require.NoError(t, codec.Decode(ctx, actualEncoding, &into, TestItemArray1Type))
+			assert.Equal(t, items, into)
+		},
+		"Returns an error if type is undefined": func(t *testing.T) {
+			item := CreateTestStruct(0, tester)
+			codec := tester.GetCodec(t)
+
+			_, err := codec.Encode(ctx, item, "NOT"+TestItemType)
+			assert.IsType(t, types.InvalidTypeError{}, err)
+
+			err = codec.Decode(ctx, []byte(""), item, "NOT"+TestItemType)
+			assert.IsType(t, types.InvalidTypeError{}, err)
+		},
+		"Returns an error encoding if arrays are the too small to encode": func(t *testing.T) {
+			if !tester.IncludeArrayEncodingSizeEnforcement() {
+				return
+			}
+
+			item1 := CreateTestStruct(0, tester)
+			items := [1]TestStruct{item1}
+			codec := tester.GetCodec(t)
+
+			_, err := codec.Encode(ctx, items, TestItemArray2Type)
+			assert.IsType(t, types.WrongNumberOfElements{}, err)
+		},
+		"Returns an error encoding if arrays are the too large to encode": func(t *testing.T) {
+			if !tester.IncludeArrayEncodingSizeEnforcement() {
+				return
+			}
+
+			item1 := CreateTestStruct(0, tester)
+			item2 := CreateTestStruct(1, tester)
+			items := [2]TestStruct{item1, item2}
+			codec := tester.GetCodec(t)
+
+			_, err := codec.Encode(ctx, items, TestItemArray1Type)
+			assert.IsType(t, types.WrongNumberOfElements{}, err)
+		},
+		"GetMaxEncodingSize returns errors for unknown types": func(t *testing.T) {
+			cr := tester.GetCodec(t)
+			_, err := cr.GetMaxEncodingSize(ctx, 10, "not"+TestItemType)
+			assert.IsType(t, types.InvalidTypeError{}, err)
+		},
+	}
+	runTests(t, tester, tests)
+}
+
+// RunChainReaderWithStrictArgsInterfaceTest is meant to be used by codecs that don't pad
+// They can assure that the right argument size is verified.
+// Padding makes that harder/impossible to verify for come codecs.
+// However, the extra verification is nice to have when possible.
+func RunChainReaderWithStrictArgsInterfaceTest(t *testing.T, tester CodecInterfaceTester) {
+	ctx := context.Background()
+	RunCodecInterfaceTests(t, tester)
+
+	tests := map[string]func(t *testing.T){
+		"Gives an error decoding extra fields on an item": func(t *testing.T) {
```

Comment:
It understands when it's run and you can right-click the test from the run list, but you can't do it before.  It generally won't know about the subtest until you run them all with interface tests anyway, since they aren't in the same file as the test that's being run.  I did a map so that I didn't duplicate tests by mistake (did that once), but I can change it to be more conventional.

Reply:
I also liked that it reduced the indent level by one this way, but convention is better to follow